### PR TITLE
WIP: Add basic unit testing for random

### PR DESCRIPTION
**Pull request by x64k**
Part of #184.

Add a basic set of unit tests for random(). Currently, this relies on
ent and performs only the most trivial tests.

While this is not sufficient to exhaustively test the quality of our
implementation, it should be reasonably sufficient to expose the most
trivial bugs that we are likely to encounter.

Signed-off-by: Alexandru Lazar <alazar@startmail.com>

### CODE REVIEW

File: test/unit/random_test.c [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+ */
+static void gen_bytestream(char *path, int len)
+{
+    int fd, err;
+    u64 buf[INTERNAL_BUFLEN / 8];
+    u64 i, nbytes = 0;
+
+    fd = open(path, O_CREAT | O_WRONLY, S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP);
+    if (fd == -1)
+    {
```
Comment (wjhun) [inline test/unit/random_test.c]: Kind of a nit, but for the sake of consistency: Could you please keep opening braces following the conditional expression?

File: test/unit/random_test.c [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+    while (nbytes < BSLEN) {
+        for (i = 0; i < INTERNAL_BUFLEN / 8; i++)
+            buf[i] = random_u64();
+
+        nbytes += INTERNAL_BUFLEN;
+
+        err = write(fd, &buf[0], INTERNAL_BUFLEN);
+        if (err == -1)
+        {
+            msg_err("error writing to bytestream file.\n");
```
Comment (wjhun) [inline test/unit/random_test.c]: Maybe print strerror(errno), and also check for EINTR here.

File: test/unit/random_test.c (hunk 1 above)
Comment (x64k) [reply test/unit/random_test.c]: Not a nit at all! I'll fix it before submitting -- sorry for the slip-up, it's the coding style I've used for the last three years or so in a full-time project and I haven't fully flushed it out of my system yet :). I'll make a local git hook with indent or something to make sure I don't end pushing it again.

File: test/unit/random_test.c (hunk 2 above)
Comment (x64k) [reply test/unit/random_test.c]: Sure thing, thanks!

**Comment by x64k** [comment]
Ah, cool, I didn't realize the CI config files are in this tree as well, I thought I'd have to pester someone to figure out how to modify the config files.

I'll push an amended commit tomorrow (I mean, well, it's already today in my timezone :) ).

Thanks for the review!

**Comment by x64k** [comment]
OK, I fixed the indenting, the missing EINTR case (doh!) and fixed the CircleCI dependencies.

I've also added a "-s" flag which, when passed, causes random_test to show the statistics it uses in human-readable format. I figured it would be useful for development/debugging -- pass/fail is useful for automated testing, but this program is also a pretty straightforward way to test the PRNG while hacking on it, when being able to see the actual values is nice. The flag is off by default, of course.

All good?

**Comment by x64k** [comment]
Awesome! I'll do the merge later today. Thanks for the review!

**Pull request merged**